contrib/dhcp_agent/dhcp_agent.py

```diff
@@ -0,0 +1,399 @@
+        )
+    fmt = logging.Formatter("[%(asctime)-12s.%(msecs)03d] "
+                            "%(levelname)-8s %(filename)s:%(lineno)d  "
+                            "%(message)s", "%Y-%m-%d %H:%M:%S")
+    handler.setFormatter(fmt)
+
+    logger.addHandler(handler)
+    return logger
+
+
+def _remove_application_lock(lockfile, logger):
+    logger.info('Removing lock')
+    os.unlink(lockfile)
+
+
+def _set_script_lock(logger):
+    lockfile = '{}.lock'.format(
+        os.path.join(tempfile.gettempdir(), os.path.split(sys.argv[0])[1])
+    )
+    f = os.open(lockfile, os.O_TRUNC | os.O_CREAT | os.O_RDWR)
+    try:
+        fcntl.lockf(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
+        os.write(f, '{}'.format(os.getpid()).encode())
+        atexit.register(_remove_application_lock, lockfile, logger)
+    except IOError as e:
+        if e.errno == errno.EAGAIN:
+            logger.critical('Script already running.')
+            sys.exit(2)
+        raise
+
+
+def _get_cmd_params_from_parser(parser):
+    """Wrapper around parser returning a dict instead of a tuple.
+
+    Returns:
+        dict: The dictonary contains pairs - option and option's value.
+    """
+    return vars(parser.parse_args()[0])
+
+
+def _check_params(params, error_callback):
+    """
+    Returns:
+        bool: True if all conditions are
+    """
+    required_params = {'host', 'key'}
+    diff = required_params - {k for k, v in params.items() if v}
+    if diff:
+        error_callback('ERROR: {} are required.'.format(
+            ', '.join(['--{}'.format(d) for d in diff]))
+        )
+        return False
+
+    dc = params.get('dc')
+    net_env = params.get('net_env')
+    if (dc and net_env) or (not dc and not net_env):
+        error_callback(
+            'ERROR: Only DC or ENV mode available.',
+        )
+        return False
+    return True
+
+
+class DHCPConfigManager(object):
+    def __init__(
+        self, logger, host, key, dc=None, net_env=None, mode=MODE_ALL,
+        verbose=False, restart=False, proto=PROTO_HTTPS,
+        dhcp_config_entries='', dhcp_config_networks='',
+        dhcp_service_name=DEFAULT_DHCP_SERVICE_NAME, **kwargs
+    ):
+        self.load_config()
+        self.logger = logger
+        self.host = host
+        self.key = key
+        self.proto = proto
+        self.can_restart_dhcp_server = restart
+        self.dhcp_service_name = dhcp_service_name
+        self.dhcp_entries_config_path = dhcp_config_entries
+        self.dhcp_networks_config_path = dhcp_config_networks
+        self.envs = net_env.split(',') if net_env else []
+        self.dcs = dc.split(',') if dc else []
+        self.mode = mode
+
+    def download_and_apply_configuration(self):
+        request_params = self.get_request_params(self.dcs, self.envs)
+        should_restart_dhcp_server = False
+        if self.mode == MODE_ALL:
+            fetch_statuses = []
+            for m in MODES_EXCLUDE_ALL:
```

Comment:
Replace it with dhcp_sections variable, which should contains all "modes" as you named it, but without "all" on the list. 
Modes is CLI term. When you parse the CLI treat "-a" explicit, without storing anywhere.